src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -317,18 +426,40 @@ protected Query doToQuery(QueryShardContext context) {
         }
 
         String indexName = context.index().getName();
-        KNNQueryFactory.CreateQueryRequest createQueryRequest = KNNQueryFactory.CreateQueryRequest.builder()
-            .knnEngine(knnEngine)
-            .indexName(indexName)
-            .fieldName(this.fieldName)
-            .vector(VectorDataType.FLOAT == vectorDataType ? this.vector : null)
-            .byteVector(VectorDataType.BYTE == vectorDataType ? byteVector : null)
-            .vectorDataType(vectorDataType)
-            .k(this.k)
-            .filter(this.filter)
-            .context(context)
-            .build();
-        return KNNQueryFactory.create(createQueryRequest);
+
+        if (k != 0) {
+            KNNQueryFactory.CreateQueryRequest createQueryRequest = KNNQueryFactory.CreateQueryRequest.builder()
+                .knnEngine(knnEngine)
+                .indexName(indexName)
+                .fieldName(this.fieldName)
+                .vector(VectorDataType.FLOAT == vectorDataType ? this.vector : null)
+                .byteVector(VectorDataType.BYTE == vectorDataType ? byteVector : null)
+                .vectorDataType(vectorDataType)
+                .k(this.k)
+                .filter(this.filter)
+                .context(context)
+                .build();
+            return KNNQueryFactory.create(createQueryRequest);
+        }
+        if (radius != null) {
+            if (!KNNEngine.getEnginesThatSupportsRadialSearch().contains(knnEngine)) {
+                throw new UnsupportedOperationException(String.format("Engine [%s] does not support radial search yet", knnEngine));
```

Comment:
remove yet.

Reply:
Why do we need remove? This is used to validate engines if support radius search in knn.

Reply:
sorry, i meant word "yet"

Reply:
Got it, updated